tests/loops/test_loops.py

```diff
@@ -713,13 +714,17 @@ def test_fit_loop_reset(tmpdir):
     epoch_loop = fit_loop.epoch_loop
     fit_loop.restarting = False
     epoch_loop.restarting = False
+    epoch_loop.val_loop.restarting = False
+    epoch_loop.val_loop.epoch_loop.restarting = False
     optimizer_loop.restarting = False
 
     # we load exactly what was saved - no reset yet
     fit_loop.load_state_dict(end_of_epoch_ckpt["loops"]["fit_loop"])
     # resetting from a end-of-epoch checkpoint SHOULD reset the current counters to 0
     fit_loop.reset()
     epoch_loop.reset()
+    epoch_loop.batch_loop.reset()
```

Comment:
shall the reset above be recursive?

Reply:
No, here we are just emulating what the loops do for testing